### PR TITLE
Add new rules for JSDoc and ES2015 style coding

### PR DESCRIPTION
**Pull request by matthewtole**
These are the rules I use in all of my JS projects; it was suggested that we might want to merge them into our master standard. Please discuss.

### CODE REVIEW

File: eslintrc.json [hunk 1]
```diff
@@ -21,6 +21,8 @@
 
   "rules": {
     "accessor-pairs": 2,
+    "arrow-body-style": [2, "as-needed"],
```
Comment (chapel) [inline eslintrc.json]: `as-needed` is the default it looks like: http://eslint.org/docs/rules/arrow-body-style

File: eslintrc.json [hunk 2]
```diff
@@ -21,6 +21,8 @@
...
+    "arrow-parens": [2, "always"],
```
Comment (chapel) [inline eslintrc.json]: `always` is the default as well http://eslint.org/docs/rules/arrow-parens

**Comment by chapel** [comment]
👍  that goes for the new rules and the PR itself.

**Comment by aheckmann-pebble** [comment]
👍

**Comment by matthewtole** [comment]
@jlai / @reyawn - I want to get agreement from whole team before merging this.

**Comment by matthewtole** [comment]
@reyawn & @jlai - As discussed in retro/sprint meeting on Monday, here's the open ESLint PR to get your feedback on.

**Comment by chapel** [comment]
@matthewtole I'd say, you should be able to merge whenever, as there has been ample time to review this. :)

**Pull request merged**